packages/editor/src/components/editor-canvas/index.js

```diff
@@ -0,0 +1,334 @@
+		'.block-editor-block-list__layout.is-root-container'
+	);
+
+	// Update type for blocks using legacy layouts.
+	const postContentLayout = useMemo( () => {
+		return layout &&
+			( layout?.type === 'constrained' ||
+				layout?.inherit ||
+				layout?.contentSize ||
+				layout?.wideSize )
+			? { ...globalLayoutSettings, ...layout, type: 'constrained' }
+			: { ...globalLayoutSettings, ...layout, type: 'default' };
+	}, [
+		layout?.type,
+		layout?.inherit,
+		layout?.contentSize,
+		layout?.wideSize,
+		globalLayoutSettings,
+	] );
+
+	// If there is a Post Content block we use its layout for the block list;
+	// if not, this must be a classic theme, in which case we use the fallback layout.
+	const blockListLayout = postContentAttributes
+		? postContentLayout
+		: fallbackLayout;
+
+	const postEditorLayout =
+		blockListLayout?.type === 'default' && ! hasPostContentAtRootLevel
+			? fallbackLayout
+			: blockListLayout;
+
+	const observeTypingRef = useTypingObserver();
+	const titleRef = useRef();
+	useEffect( () => {
+		if ( ! autoFocus || ! isCleanNewPost() ) {
+			return;
+		}
+		titleRef?.current?.focus();
+	}, [ autoFocus, isCleanNewPost ] );
+
+	// Add some styles for alignwide/alignfull Post Content and its children.
+	const alignCSS = `.is-root-container.alignwide { max-width: var(--wp--style--global--wide-size); margin-left: auto; margin-right: auto;}
+		.is-root-container.alignwide:where(.is-layout-flow) > :not(.alignleft):not(.alignright) { max-width: var(--wp--style--global--wide-size);}
+		.is-root-container.alignfull { max-width: none; margin-left: auto; margin-right: auto;}
+		.is-root-container.alignfull:where(.is-layout-flow) > :not(.alignleft):not(.alignright) { max-width: none;}`;
+
+	return (
+		<>
+			{ themeSupportsLayout &&
+				! themeHasDisabledLayoutStyles &&
+				renderingMode === 'post-only' && (
+					<>
+						<LayoutStyle
+							selector=".editor-editor-canvas__post-title-wrapper"
+							layout={ fallbackLayout }
+						/>
+						<LayoutStyle
+							selector=".block-editor-block-list__layout.is-root-container"
+							layout={ postEditorLayout }
+						/>
+						{ align && <LayoutStyle css={ alignCSS } /> }
+						{ postContentLayoutStyles && (
+							<LayoutStyle
+								layout={ postContentLayout }
+								css={ postContentLayoutStyles }
+							/>
+						) }
+					</>
+				) }
+			{ renderingMode === 'post-only' && (
+				<div
+					className={ classnames(
+						'editor-editor-canvas__post-title-wrapper',
+						// The following class is only hear for backward comapatibility
+						// some themes might be using it to style the post title.
+						'edit-post-visual-editor__post-title-wrapper',
+						{
+							'has-global-padding': hasRootPaddingAwareAlignments,
+						}
+					) }
+					contentEditable={ false }
+					ref={ observeTypingRef }
+					style={ {
+						// This is using inline styles
+						// so it's applied for both iframed and non iframed editors.
+						marginTop: '4em',
```

Comment:
In the post editor, on `trunk` this is `4rem` instead of `4em`, so in testing manually there is a slight yet subtle difference. Not a blocker at all, just thought I'd mention it in case it's something you wanted to tweak in a follow-up (I'm not sure which one's preferred).

Reply:
If I had to guess, I'd say `rem` is preferred because it's relative to `:root` and not influenced by the element's parent(s)? Normally it scales more uniformly with `rem` units.